QueryBuilder/Compilers/SnowflakeCompiler.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Globalization;
+
+
+namespace SqlKata.Compilers
+{
+    // Snowflake-specific compiler. The SQL syntax for Snowflake is actually
+    // fairly similar with Postgres.
+    public sealed class SnowflakeCompiler : PostgresCompiler
+    {
+        public override string EngineCode { get; } = EngineCodes.Snowflake;
+
+        protected override string LastId =>
+            throw new NotSupportedException($"LastId not supported in {EngineCode} compiler");
+
+        public override string Parameter(SqlResult ctx, object parameter)
+        {
+            switch (parameter)
+            {
+                case DateTimeOffset dto:
+                    {
+                        // DateTimeOffset is always w.r.t. UTC
+                        ctx.Bindings.Add(dto.ToUniversalTime().ToString("u", CultureInfo.InvariantCulture));
+                        return "?";
+                    }
+
+                case DateTime dt:
+                    {
+                        if (dt.Kind == DateTimeKind.Unspecified)
+                        {
+                            throw new ArgumentException("DateTime kind must be explicitly specified for the Snowflake compiler");
+                        }
+                        if (dt.Kind == DateTimeKind.Local)
+                        {
+                            // If we ever want to support this, we first need a
+                            // way to test it which can deal with the unknown
+                            // local time offset.
+                            throw new ArgumentException("DateTime kind 'Local' is not supported for the Snowflake compiler");
+                        }
+                        ctx.Bindings.Add(dt.ToUniversalTime().ToString("u", CultureInfo.InvariantCulture));
+                        return "?";
```

Comment:
seems like it might be better to use a switch here for the 3 possible kinds (Unspecified/Utc/Local)

Reply:
pushed an update